packages/labs/eleventy-plugin-lit/src/index.ts

```diff
@@ -20,35 +31,36 @@ module.exports = {
     eleventyConfig.addTransform(
       'render-lit',
       async (content: string, outputPath: string) => {
-        if (outputPath.endsWith('.html')) {
-          const render = (
-            await import('@lit-labs/ssr/lib/render-with-global-dom-shim.js')
-          ).render;
-          const html = (await import('lit')).html;
-          const unsafeHTML = (await import('lit/directives/unsafe-html.js'))
-            .unsafeHTML;
-          // TODO(aomarks) This doesn't work in Eleventy --watch mode, because
-          // the ES module cache never gets cleared. Switch to isolated VM
-          // modules so that we can control this.
-          await Promise.all(
-            (options.componentModules ?? []).map(
-              (module) => import(path.resolve(process.cwd(), module))
-            )
-          );
-          let head, body, tail;
-          const page = content.match(/(.*)(<body.*<\/body>)(.*)/);
-          if (page) {
-            [head, body, tail] = page;
-          } else {
-            head = `<html><head></head>`;
-            body = `<body>${content}</body>`;
-            tail = `</html>`;
-          }
-          return (
-            head + iterableToString(render(html`${unsafeHTML(body)}`)) + tail
-          );
+        if (!outputPath.endsWith('.html')) {
+          return content;
         }
-        return content;
+
+        const render = (
+          await import('@lit-labs/ssr/lib/render-with-global-dom-shim.js')
+        ).render;
+        const html = (await import('lit')).html;
+        const unsafeHTML = (await import('lit/directives/unsafe-html.js'))
+          .unsafeHTML;
+        // TODO(aomarks) This doesn't work in Eleventy --watch mode, because
+        // the ES module cache never gets cleared. Switch to isolated VM
+        // modules so that we can control this.
+        await Promise.all(
+          (options.componentModules ?? []).map(
+            (module) => import(path.resolve(process.cwd(), module))
+          )
+        );
+
+        const rendered = iterableToString(render(html`${unsafeHTML(content)}`));
+        // Lit SSR includes special comment markers around templates to identify
+        // them during hydration. However, it's not possible for the top-level
+        // template that we've just rendered to ever be hydrated, because it was
+        // only defined ephemerally right now. So the comments serve no purpose
+        // in this case and can be removed. This also avoids the issue of
+        // comments appearing before <!doctype> declarations.
+        const outerMarkersTrimmed = rendered
```

Comment:
Rather than removing the markers after they're already emitted, could we add an option to SSR to not render the markers?

Reply:
Started on this, but the issue I have hit is that there are two layers of markers:

1. For the outer template
2. For the `unsafeHTML` directive

So we can add an `outerMarkers:false` option to `render`, but that's only going to handle layer 1 and not 2. We don't want the option it to be generally recursive, since that would break all hydration. I suppose it could take a `depth` parameter, and only omit comments up to that depth.

Once we switch to vm modules, I think we won't have the `unsafeHTML` directive anymore, so this problem will go away.

Reply:
seems fine for now, but I do wonder if we're going to want ways to control marker output more generally if we ever have SSR-only rendering.

Reply:
Maybe a `markerStartDepth` parameter would work. `0` by default, `1` if you don't care about top-level, `2` for top-level + e.g. `unsafeHtml`, `-1` to skip all?